### PR TITLE
website: Add TOC links in API sidebar

### PR DESCRIPTION
**Pull request by backspace**
This is minimal-functionality API TOC implementation, here’s a GIF:

api-toc-demonstration

I only added the TOC to pages where it made sense; `search` and `regions` don’t have it, for instance. ~The overview (`index`) does have it but I’m not sure whether that’s suitable.~

The UX would be better if the sidebar were fixed and/or clicking links to different pages maintained the scroll position, but since there’s work underway to replace these sites, I didn’t want to put much more time into this. I think it might also be preferable to have the links be like `GET /v1/jobs` etc, but that’s much harder to implement.

You can see it live [here](https://deploy-preview-6109--nomad-ui.netlify.com/api/index.html). That preview is produced by the changes to `netlify.toml`, which should be removed if this is to be merged.

ETA: I chose to put the TOC in the sidebar rather than at the top of the main content, but that’s another possibility if it seems preferable. It would make it easier to access the TOC for pages that are lower down in the listing.

### CODE REVIEW

**Comment by backspace** [comment]
hmm I’ve incorrectly branched off `stable-website` I see now 😢

File: website/source/layouts/api.erb [hunk 1]
```diff
@@ -3,6 +3,7 @@
     <ul class="nav docs-sidenav">
       <li<%= sidebar_current("api-overview") %>>
         <a href="/api/index.html">Overview</a>
+        <%= api_toc("index") %>
```
Comment (schmichael) [inline website/source/layouts/api.erb]: Let's remove the TOC on the index. It's nice, but it pushes the main API endpoint docs down page pretty far and they're what people are probably looking for.

Comment (backspace) [reply website/source/layouts/api.erb]: yeah, I was wondering about that, will do, thanks!

**Comment by backspace** [comment]
I keep thinking about it and am maybe talking myself into the TOCs being in the main content area after the title and initial text, before the first subhead. This would mean it would be more consistently visible after any navigation, whereas the current implementation has the TOC wherever the page sits in the sidebar, which can sometimes be several viewports down.

So, things to consider.

**Comment by backspace** [comment]
I’m closing this in favour of #6113.

**Pull request closed without merging**